salt/_states/metalk8s_kubernetes.py

```
@@ -1,1938 +1,164 @@
-
-    return ret
-
-
-def namespace_annotation_present(
-        name,
-        annotation_key,
-        annotation_value,
-        **kwargs):
-    '''
-    Ensures that the namespace is patched.
-
-    name
-        The name of the namespace
-
-    annotation_key
-        Key of the annotation
-    annotation_value
-        Value of the annotation
-    '''
-
-    ret = {'name': name,
-           'changes': {},
-           'result': False,
-           'comment': ''}
-
-    annotations = __salt__['metalk8s_kubernetes.namespace_annotations'](
-        name, **kwargs
-    ) or {}
-
-    if annotation_key not in annotations:
-        ret['comment'] = 'The annotation is going to be set'
-        if __opts__['test']:
-            ret['result'] = None
-            return ret
-
-        __salt__['metalk8s_kubernetes.namespace_add_annotation'](
-            name=name,
-            annotation_key=annotation_key,
-            annotation_value=annotation_value,
-            **kwargs
-        )
-
-    elif annotations[annotation_key] == annotation_value:
-        ret['result'] = True
-        ret['comment'] = (
-            'The annotation is already set and has the specified value'
+        ret['comment'] = 'The object is going to be {}'.format(
+            'created' if obj is None else 'replaced'
         )
         return ret
 
-    else:
-        if __opts__['test']:
-            ret['result'] = None
-            ret['comment'] = 'The annotation is going to be updated'
-            return ret
-
-        ret['comment'] = 'The annotation is already set, changing the value'
-
-        __salt__['metalk8s_kubernetes.namespace_add_annotation'](
-            name=name,
-            annotation_key=annotation_key,
-            annotation_value=annotation_value,
+    if obj is None:
+        __salt__['metalk8s_kubernetes.create_object'](
+            name=name_arg, manifest=manifest,
             **kwargs
         )
+        ret['changes'] = {'old': 'absent', 'new': 'present'}
```

So we bubble up any exceptions `create_object` could raise?

---

Salt will automatically raise this exception directly so for me it's good enough, except if we really want to hide exception but I think it's not a good idea 
state return will look like this (don't take care of the error message, was only a typo :stuck_out_tongue: )
```
metalk8s_kubernetes_|-toto_|-toto_|-object_updated:
    ----------
    __id__:
        toto
    __run_num__:
        0
    __sls__:
        None
    changes:
        ----------
    comment:
        An exception occurred in this state: Traceback (most recent call last):
          File "/usr/lib/python2.7/site-packages/salt/state.py", line 1919, in call
            **cdata['kwargs'])
          File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1918, in wrapper
            return f(*args, **kwargs)
          File "/var/cache/salt/master/extmods/states/metalk8s_kubernetes.py", line 136, in object_updated
            diff = __utils__['dictdiffer.recursive_diff'](obj, new_obj.to_dict())
        AttributeError: 'dict' object has no attribute 'to_dict'
    duration:
        35.622
    name:
        toto
    result:
        False
    start_time:
        14:08:48.872329
```